src/status_im2/common/standard_authentication/view.cljs

```diff
@@ -0,0 +1,69 @@
+(ns status-im2.common.standard-authentication.view
+  (:require
+   [quo2.core :as quo]
+   [quo2.theme :as quo.theme]
+   [reagent.core :as reagent]
+   [utils.re-frame :as rf]
+   [react-native.touch-id :as biometric]
+   [utils.i18n :as i18n]
+   [status-im2.common.standard-authentication.enter-password.view :as enter-password]
+   [react-native.core :as rn]))
+
+(defn authorize
+  [{:keys [on-enter-password biometric-auth? on-auth-success on-auth-fail fallback-button-label theme
+           on-close]}]
+  (biometric/get-supported-type
+   (fn [biometric-type]
+     (if (and biometric-auth? biometric-type)
+       (biometric/authenticate
+        {:reason     (i18n/label :t/biometric-auth-confirm-message)
+         :on-success (fn [response]
+                       (on-auth-success)
+                       (js/console.log "response" response))
+         :on-fail    (fn [error]
+                       (js/console.log "Authentication Failed. Error:" error)
+                       (on-auth-fail)
+                       (rf/dispatch [:show-bottom-sheet
+                                     {:theme   theme
+                                      :content (fn []
+                                                 [enter-password/view
+                                                  {:on-enter-password on-enter-password}])}]))})
+
+       (rf/dispatch [:show-bottom-sheet
+                     {:on-close on-close
+                      :theme    theme
+                      :content  (fn []
+                                  [enter-password/view
+                                   {:on-enter-password on-enter-password
+                                    :button-label      fallback-button-label}])}])))))
+
+(defn- view-internal
+  [{:keys [biometric-auth?
+           track-text
+           customization-color
+           fallback-button-label
+           on-enter-password
+           on-auth-success
+           on-auth-fail
+           size
+           theme]
+    :or   {on-auth-success #() on-auth-fail #()}}]
```

Comment:
It's a bit weird to define empty anonymous functions like this. This might be a sign that the code using these functions in `authorize` is not checking for function existence, which is usually recommended, otherwise it's too easy to have an exception.

In the `authorize` function, we can change it to check the callback functions exist before calling them. The pattern is `(when on-auth-success (on-auth-success))`. Also, the default destructuring doesn't protect the code from the caller of `view` explicitly passing `nil` as the calllback (which can be surprisingly easy to happen in Clojure).

Reply:
yep, I agree. will sort 👍